l1-contracts/contracts/governance/ServerNotifier.sol

```diff
@@ -0,0 +1,38 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.0;
+
+import {Ownable2Step} from "@openzeppelin/contracts-v4/access/Ownable2Step.sol";
+import {ZeroAddress} from "../common/L1ContractErrors.sol";
+import {ReentrancyGuard} from "../common/ReentrancyGuard.sol";
+
+contract ServerNotifier is Ownable2Step, ReentrancyGuard {
+    mapping(address chainAdmin => uint256 chainId) public registeredChains;
```

Comment:
you can take a look at how it is done in ValidatorTimelock (we can query chain admin directly from the chain and there would be no need to e.g. register the chain in this case)

Reply:
(I understand that the contract is WIP, just left the comment for the future)

Reply:
good suggestion, fixes a lot of troubles